utils/firestore.js

```diff
@@ -3539,6 +3539,196 @@ const updateParticipantCorrection = async (participantData) => {
+                                        incentiveEligible = true;
+                                    }
+                                });
+                            });
+                        }
+                    }
+                }
+            }
+        }
+        return { 
+            isEligibleForIncentive: incentiveEligible, 
+            participantData 
+        };
+    } catch (error) {
+        console.error('Error checking participant for eligible incentive:', error);
+        throw {
+            code: 500,
+            message: `Error checking participant for eligible incentive. ${error.message}`
+        };
+    }
+};
+
+/**
+ * Update participant incentive eligibility for NORC Incentive Eligibility tool
+ * @param {string} connectId - Connect ID of the participant
+ * @param {string} currentPaymentRound - Payment round to update eligibility for participant
+ * @param {string} dateOfEligibility - Date of eligibility for incentive ISO 8601 format
+ * @returns {object} - Updated participant document
+*/
+const updateParticipantIncentiveEligibility = async (connectId, currentPaymentRound, dateOfEligibility) => { 
+    try {
+        const { paymentRound, eligibleForIncentive, yes, no, norcPaymentEligibility, timestampPaymentEligibilityForRound } = fieldMapping;
+
+        const eligibilityCheck = await checkParticipantForEligibleIncentive(connectId, currentPaymentRound);
+        const isEligibleForIncentive = eligibilityCheck.isEligibleForIncentive;
+        if (!isEligibleForIncentive) throw { code: 400, message: 'Participant is not eligible for incentive update.' };
+
+        const participantData = eligibilityCheck.participantData;
+        const currentPaymentRoundName = currentPaymentRound; // baseline or future payment rounds
+
+        const snapshot = await db.collection('participants').where('Connect_ID', '==', connectId).get();
+        if (snapshot.empty) throw { code: 404, message: 'Participant not found.' };
+
+        const participantRef = snapshot.docs[0].ref;
+
+        if (participantData[paymentRound][currentPaymentRoundName][eligibleForIncentive] === no) {
+            console.log("🚀 ~ updateParticipantIncentiveEligibility ~ connectId, paymentRound, dateOfEligibility:", connectId, currentPaymentRound, dateOfEligibility)
+            await participantRef.update({
+                [`${paymentRound}.${currentPaymentRoundName}.${eligibleForIncentive}`]: yes,
+                [`${paymentRound}.${currentPaymentRoundName}.${norcPaymentEligibility}`]: yes,
+                [`${paymentRound}.${currentPaymentRoundName}.${timestampPaymentEligibilityForRound}`]: dateOfEligibility
+            });
+            const updatedDoc = await participantRef.get();
+            if (!updatedDoc.exists) throw { code: 404, message: 'Updated document not found.' }
```

Comment:
Multiple non-400 errors here with an error message set, but also just a `throw error`, which will result in inconsistent logging.